backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/LinkService.cs

```diff
@@ -0,0 +1,82 @@
+﻿using AutoMapper;
+using EasyMeets.Core.BLL.Interfaces;
+using EasyMeets.Core.Common.DTO.Team;
+using EasyMeets.Core.DAL.Context;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using System.Text;
+using System.Web;
+
+namespace EasyMeets.Core.BLL.Services
+{
+    public class LinkService : BaseService, ILinkService
+    {
+        private readonly IConfiguration _config;
+        private string _applicationUri;
+        public LinkService(EasyMeetsCoreContext context, IMapper mapper, IConfiguration configuration) : base(context, mapper)
+        {
+            _config = configuration;
+            _applicationUri = _config["ApplicationUri"];
+        }
+
+        public async Task<string> GenerateNewPageLinkAsync(long teamId, string teamName)
+        {
+            if (!await _context.Teams.AnyAsync(t => t.Id != teamId && t.PageLink == teamName))
+            {
+                return teamName;
+            }
+            int index = 1;
+
+            while (await _context.Teams.AnyAsync(t => t.Id != teamId && t.PageLink == $"{teamName}{index}"))
```

Comment:
It can be simplified to .CountAsync() method with predicate
![image](https://user-images.githubusercontent.com/46250167/191508320-8858deb0-17a1-4229-b261-3415c338bb6a.png)